### PR TITLE
DEVTOOLS: added script to create steam achievement table

### PR DESCRIPTION
**Pull request by mgerhardy**
This script automatically extracts the needed entries for the steam achievements from e.g. https://steamdb.info/app/212050/stats/

It currently uses the macros found in the AGS `achivements_tables.h` file.

Just give it the steam game id as parameter followed by the scummvm game id

Also see this PR: https://github.com/scummvm/scummvm/pull/2815

We could even auto create the full list of achievements if we could look up the detection entries for steam game ids.

`./steam_achivements.py 212050 resonance`

Example output:

```
 ~/dev/scummvm$  ./devtools/steam_achivements.py                                                                                                                                                                                                     
missing steam game id as first parameter - assuming 212050
found 38 achievements
missing scummvm game id - assuming Resonance · AppID: 212050
{
	"Resonance · AppID: 212050",
	Common::STEAM_ACHIEVEMENTS,
	"212050",
	{
		ACHIEVEMENT_SIMPLE_ENTRY("THATWENTWELL", "That went well", "Talk to that nice girl on the subway"),
		ACHIEVEMENT_SIMPLE_ENTRY("MIRRORMASTER", "Mirror Master", "Grow some eyes in the back of your head"),
		ACHIEVEMENT_SIMPLE_ENTRY("GOTTAHIDE", "Gotta hide", "Escape a nightmare"),
		ACHIEVEMENT_SIMPLE_ENTRY("GETTHESCOOP", "Get the scoop", "The pulitzer is in the bag"),
		ACHIEVEMENT_SIMPLE_ENTRY("JUSTBECAUSE", "Just because", "Ray's kind of a jerk"),
		ACHIEVEMENT_SIMPLE_ENTRY("THATGUY", ""That" guy", "That creep on the subway? Yeah, that's you."),
		ACHIEVEMENT_SIMPLE_ENTRY("DUO", "Duo", "Gain control of two characters"),
		ACHIEVEMENT_SIMPLE_ENTRY("TEAMWORK", "Teamwork", "Team up to fix the lab's water system"),
		ACHIEVEMENT_SIMPLE_ENTRY("SAVIOR", "Savior", "Save Dr. Morales"),
		ACHIEVEMENT_SIMPLE_ENTRY("CUTTHEROPE", "Cut the rope", "Save yourself by destroying a rope"),
		ACHIEVEMENT_SIMPLE_ENTRY("GOODLISTENER", "Good listener", "Listen to Saul's whole story without interrupting him"),
		ACHIEVEMENT_SIMPLE_ENTRY("NOSMOKING", "No smoking", "Discover the chemisty department's secret"),
		ACHIEVEMENT_SIMPLE_ENTRY("WIRED", "Wired", "Rewire the lab door circuit"),
		ACHIEVEMENT_SIMPLE_ENTRY("ANGEL", "Angel", "Open Dr. Morales's journal without damaging it"),
		ACHIEVEMENT_SIMPLE_ENTRY("DREAMMASTER", "Dream master", "Get all the memories from Anna's dream maze"),
		ACHIEVEMENT_SIMPLE_ENTRY("QUARTET", "Quartet", "Gain control of all four characters"),
		ACHIEVEMENT_SIMPLE_ENTRY("BLUEPRINTSNATCHER", "Blueprint snatcher", "Nab the blueprints from the police archive"),
		ACHIEVEMENT_SIMPLE_ENTRY("SEETHEWIZARD", "See the Wizard", "Find Morales's financial information"),
		ACHIEVEMENT_SIMPLE_ENTRY("MAGNETICPERSONALITY", "Magnetic personality", "Maneuver everyone through the super collider"),
		ACHIEVEMENT_SIMPLE_ENTRY("ACCESSGRANTED", "Access granted", "Gain access to the inner vault"),
		ACHIEVEMENT_SIMPLE_ENTRY("IPROMISEYOU", "I promise you", "Your uncle holds the key"),
		ACHIEVEMENT_SIMPLE_ENTRY("SAFECRACKER", "Safecracker", "Crack Tortoise's safe"),
		ACHIEVEMENT_SIMPLE_ENTRY("BLACKMAILER", "Blackmailer", "Gain Tortoise's help by resorting to blackmail"),
		ACHIEVEMENT_SIMPLE_ENTRY("ARCHIVEMASTER", "Archive master", "Get the blueprints on your first try"),
		ACHIEVEMENT_SIMPLE_ENTRY("HIPPOCRATES", "Hippocrates", "Pay Ozzy while keeping your professional integrity intact"),
		ACHIEVEMENT_SIMPLE_ENTRY("CLOSETOHEART", "Close to heart", "Learn Bennet's secret"),
		ACHIEVEMENT_SIMPLE_ENTRY("FINDINGNORTH", "Finding north", "Use a compass to find a new location"),
		ACHIEVEMENT_SIMPLE_ENTRY("KEYNABBER", "Key nabber", "Remove the contents of a locker without opening it"),
		ACHIEVEMENT_SIMPLE_ENTRY("PHONETRACER", "Phone tracer", "Track the movements of a killer"),
		ACHIEVEMENT_SIMPLE_ENTRY("MAPMASTER", "Technophile", "Find a new location on the map using fancy technology"),
		ACHIEVEMENT_SIMPLE_ENTRY("LESSEROFTWOEVILS", "Lesser of two evils", "You sided with an enemy to stop a greater foe"),
		ACHIEVEMENT_SIMPLE_ENTRY("THENEWWORLDORDER", "The new world order", "You stopped a killer, but let the big fish go free"),
		ACHIEVEMENT_SIMPLE_ENTRY("INDEPENDENT", "Independent", "Finish the game without asking another character for a hint"),
		ACHIEVEMENT_SIMPLE_ENTRY("LISTENTOUS", "Listen to us", "Play the entire game with commentary mode active"),
		ACHIEVEMENT_SIMPLE_ENTRY("HIGHSCORE", "High Score", "SUPER ACHIEVEMENT: Finish the game with a full score"),
		ACHIEVEMENT_SIMPLE_ENTRY("SAFESEER", "Safe seer", "INSANE ACHIEVEMENT: Open Tortoise's safe without using any "visual aids""),
		ACHIEVEMENT_SIMPLE_ENTRY("QUICKESCAPE", "Quick escape", "INSANE ACHIEVEMENT: Escape from the monster without locking the door"),
		ACHIEVEMENT_SIMPLE_ENTRY("DECODER", "Decoder ring", "SUPER ACHIEVEMENT: Decode a journal"),
		ACHIEVEMENTS_LISTEND
	}
},
```

### CODE REVIEW

File: devtools/steam_achivements.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#
+# It will extract the steam stats information to create the scummvm achievement tables with the macros found
+# in AGS achievements_table.h
+
+import requests
+from requests_html import HTMLSession
+import sys
+
+steam_game_id = sys.argv[1:]
+scummvm_game_id = sys.argv[2:]
```
Comment (mduggan) [inline devtools/steam_achivements.py:16]: these probably should be below the length check below?

Comment (mgerhardy) [reply devtools/steam_achivements.py:16]: would look saner, yes - but it works like this due to the `:`

Comment (mgerhardy) [reply devtools/steam_achivements.py]: Fixed

**Comment by mgerhardy** [comment]
https://gist.github.com/mgerhardy/6648eed972bca7ed2d70574b0be3e180#file-steam_ags_achievements-sh

**Pull request merged**